File: frontend/src/types/nota.ts
export interface Nota {
  id: number;
  temaId: number;
  titulo: string;
  conteudo: string;
  resumoIa?: string | null;
  dataCriacao?: string | null;
  dataAtualizacao?: string | null;
  nomeTema?: string | null;
}

export interface NotaRequestDto {
  titulo: string;
  conteudo: string;
  temaId: number;
  resumoIa?: string | null;
}

export interface NotaUpdateDto {
  titulo: string;
  conteudo: string;
  temaId: number;
  resumoIa?: string | null;
}

export interface NotaResponseDto {
  id: number;
  titulo: string;
  conteudo: string;
  resumoIa?: string | null;
  dataCriacao: string;
  temaId: number;
  nomeTema?: string | null;
}

export interface CreateNotaDto {
  titulo: string;
  conteudo: string;
  temaId: number;
  resumoIa?: string | null;
}

File: backend/StudyFlow.Api/Domain/Entities/Nota.cs

namespace StudyFlow.Api.Domain.Entities
{
    public class Nota
    {
        public int Id { get; set; }
        public string? Titulo { get; set; }
        public string? Conteudo { get; set; }
        public string? ResumoIA { get; set; }
        public DateTime DataCriacao { get; set; } = DateTime.UtcNow;

        public Guid UsuarioId { get; set; }
        public Usuario? Usuario { get; set; }

        public int? TemaId { get; set; }
        public Tema? Tema { get; set; }

        public ICollection<ConexaoNota> ConexoesOrigem { get; set; } = new List<ConexaoNota>();
        public ICollection<ConexaoNota> ConexoesDestino { get; set; } = new List<ConexaoNota>();
    }
}